packages/api-server/api_server/gateway.py

```diff
@@ -248,6 +270,62 @@ def handle_delivery_alert(delivery_alert: DeliveryAlert):
         )
         self._subscriptions.append(delivery_alert_request_sub)
 
+        def convert_fleet_alert(fleet_alert: RmfFleetAlert):
+            tier = AlertRequest.Tier.Info
+            if fleet_alert.tier == RmfFleetAlert.TIER_WARNING:
+                tier = AlertRequest.Tier.Warning
+            elif fleet_alert.tier == RmfFleetAlert.TIER_ERROR:
+                tier = AlertRequest.Tier.Error
+
+            parameters = []
+            for p in fleet_alert.alert_parameters:
+                parameters.append(AlertParameter(name=p.name, value=p.value))
+
+            return AlertRequest(
+                id=fleet_alert.id,
+                unix_millis_alert_time=round(datetime.now().timestamp() * 1000),
+                title=fleet_alert.title,
+                subtitle=fleet_alert.subtitle,
+                message=fleet_alert.message,
+                display=fleet_alert.display,
+                tier=tier,
+                responses_available=fleet_alert.responses_available,
+                alert_parameters=parameters,
+                task_id=fleet_alert.task_id if len(fleet_alert.task_id) > 0 else None,
+            )
+
+        def handle_fleet_alert(fleet_alert: AlertRequest):
+            logging.info("Received fleet alert:")
+            logging.info(fleet_alert)
+
+            # Handle request for checking all location completion success for
+            # this task
+            is_final_check = is_final_location_alert_check(fleet_alert)
+            if is_final_check:
+                successful_so_far = task_id_to_all_locations_success_cache.lookup(
+                    fleet_alert.task_id
+                )
+                if successful_so_far is None or not successful_so_far:
+                    self.respond_to_alert(fleet_alert.id, LocationAlertFailResponse)
```

Comment:
1. api-server should not be sending respond by itself.
2. The alert is sent without writing to db.

Why does the FA need to ask the api-server if previous alerts are successful? I guess the reason for this is that the FA does not keep state between phases? In any case, this should done in SCAS, the behavior seems very specific.